lib/util/stat.js

```diff
@@ -42,6 +44,14 @@ function checkPaths (src, dest, funcName, opts, cb) {
 
     if (destStat) {
       if (areIdentical(srcStat, destStat)) {
+        const srcBaseName = path.basename(src)
+        const destBaseName = path.basename(dest)
+        if (funcName === 'move' &&
+          isCaseInsensitiveSystem &&
```

Comment:
perhaps better to simply not test for this. If the `stat()` result is the same, it's the same file.

Reply:
... And in that case, just call rename(). I can't think of a case where it would fail

Reply:
That makes sense! I am ok with not checking for case sensitivity of the system as this is not comprehensive enough!